### DISCUSSION TITLE
Catching `HttpTransportException` when emails fail to send

### DISCUSSION THREAD

**Discussion by atymic** [laravel/framework#46862] [Q&A]
Hey all!

We're having a bit of an issue with email sending, namely that the postmark driver throws exceptions in cases were we try to send to someone who has previously bounced. This could be useful, but there's no way to catch and process these on an application level. 

This mean that when sending an email in a queued job (sending sync, since we are already on the queue) this exception terminates the job, where really this should let it continue.

Has anyone come up with a way of dealing with this? Our current solution is to just queue the emails as well, but then they are retried because of aforementioned exception being thrown 

Thanks!

**Comment by knorthfield** [comment]
We have the same problem. If someone unsubscribes and we try to send an email then our whole app dies! Very inconvienient.

**Comment by mstaack** [comment]
unfortunately mailables do not support job middlwares.
maybe hook into messagesending event?

haven't found a solution so far..

**Reply by mstaack** [reply]
ok i was wrong, mailables do support middlwares. this works just great:

`Mailable`:
```
    public function middleware(): array
    {
        return [
            new IgnorePostmarkSuppressions(),
        ];
    }
```

and a middlware like:
```
class IgnorePostmarkSuppressions
{
    // unfortunately the postmark api response code for this error (406) is only in the message string
    public function handle(SendQueuedMailable $mailable, Closure $next): void
    {
        try {
            $next($mailable);
        } catch (HttpTransportException $exception) {
            throw_unless($this->isSuppressed($exception), $exception);
        }
    }

    private function isSuppressed(HttpTransportException $exception): bool
    {
        return str_contains($exception->getMessage(), '(code 406)');
    }
```
